fdbserver/DataDistribution.actor.cpp

```diff
@@ -4486,6 +4514,45 @@ ACTOR Future<Void> ddSnapCreate(DistributorSnapRequest snapReq, Reference<AsyncV
 	return Void();
 }
 
+ACTOR Future<Void> ddExclusionSafetyCheck(DistributorExclusionSafetyCheckRequest req,
+                                          Reference<DataDistributorData> self, Database cx) {
+	TraceEvent("DDExclusionSafetyCheckBegin");
+	vector<StorageServerInterface> ssis = wait(getStorageServers(cx));
+	DistributorExclusionSafetyCheckReply reply(true);
+	if (!self->teamCollection) {
+		TraceEvent("DDExclusionSafetyCheckTeamCollectionInvalid");
+		reply.safe = false;
+		req.reply.send(reply);
+		return Void();
+	}
+	vector<UID> excludeServerIDs;
+	// Go through storage server interfaces and translate Address -> server ID (UID)
```

Comment:
What if DBA excludes an address not exist? 
I think we should return false.

Reply:
Excluding addresses that don't correspond to valid servers is currently allowed on master, and there are no checks around that. Would you suggest a system-wide change or just special behaviour for our safety check here?

Reply:
I see. In this case, the check for non-exist address does not have to be done in this PR.
Would you mind creating an issue to track it?
We want two properties:
1) we don't want to add a non-exist address to the excludeServer list
2) we prefer to warn/reject DBA's request to exclude a non-exist address.

Reply:
There are potential use cases for excluding something that doesn't exist. For example, you may want to exclude a host prior to adding it to the cluster to prevent any recruitments happening before you want them. 

It's also possible that other potential features may provide this functionality, as I think I've heard various interest in being able separate the steps of adding hosts and using them.

Reply:
Fair enough. I didn't know those examples.